### PR TITLE
Collect all weather data from the Gateway

### PR DESCRIPTION
**Pull request by bouwew**
@CoMPaTech I was preparing a little bit for the Anna P1 device and I realized there is also interesting data available at the Gateway / Home location for thermostat devices.

Also, this will allow us to add the weather-platform to the Integrations.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new `weather` data structure for devices, replacing the previous `sensors` structure, which includes attributes like humidity, outdoor temperature, solar irradiance, weather description, wind bearing, and wind speed.
	- Updated item counts for multiple gateways, reflecting the addition of new items being tracked.

- **Bug Fixes**
	- Adjusted expected values in test cases to align with the new data structures and item counts.

### CODE REVIEW

**Comment by CoMPaTech** [comment]
I think we came across this earlier, but decided (back then) not to incorporate it as it was also just generic data from the internet (or maybe from a Core member, I don't really remember) - then again, it is provided data and as such could at least mimic what, for instance Anna, displays.

File: plugwise/constants.py [hunk 1]
```diff
@@ -385,6 +391,15 @@
...
 }
+WeatherType = Literal[
+    "humidity",
+    "outdoor_temperature",
+    "solar_irradiance",
+    "weather_description",
+    "wind_bearing",
+    "wind_speed",
+]
+WEATHER: Final[tuple[str, ...]] = get_args(WeatherType)
```
[the remark this answers is not in the document]
Comment (CoMPaTech) [reply plugwise/constants.py:402]: That might make sense, unless one is from the Smile and the other the HA standardised form?

**Review by CoMPaTech — COMMENTED** [review]
While reviewing, what do we do with outdoor_temperature if one has both? I remember we decided to opt for the physical sensor instead of the perceived one, just to be clear that should be (also) obvious to end users.

**Comment by bouwew** [comment]
> While reviewing, what do we do with outdoor_temperature if one has both? I remember we decided to opt for the physical sensor instead of the perceived one, just to be clear that should be (also) obvious to end users.

We solved that a long time ago by renaming it to `outdoor_air_temperature`.

**Comment by bouwew** [comment]
Closing this PR. It's doesn't really add any value.
Also, these updates are not helpful in preparing for the new Anna P1 device.

**Pull request closed without merging**